File: FinalBootMVC/src/main/java/com/springboot/pkg/model/RH.java
package com.springboot.pkg.model;

import javax.persistence.DiscriminatorColumn;
import javax.persistence.DiscriminatorType;
import javax.persistence.Embedded;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.Inheritance;
import javax.persistence.InheritanceType;
import javax.persistence.SequenceGenerator;
import javax.persistence.Version;

import com.fasterxml.jackson.annotation.JsonSubTypes;
import com.fasterxml.jackson.annotation.JsonTypeInfo;
import com.fasterxml.jackson.annotation.JsonView;
import com.fasterxml.jackson.annotation.JsonSubTypes.Type;
import com.springboot.pkg.model.jsonview.JsonViews;

@Entity
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "type")
@JsonSubTypes({ @Type(value = Ordinateur.class, name = "stagiaire"), @Type(value = Salle.class, name = "formateur"),
		@Type(value = Videoprojecteur.class, name = "gestionnaire"), @Type(value = Videoprojecteur.class, name = "technicien")})
@Inheritance(strategy = InheritanceType.SINGLE_TABLE)
@DiscriminatorColumn(name = "type", discriminatorType = DiscriminatorType.STRING)
public abstract class RH {

	@Id
	@SequenceGenerator(name = "RhSeq", sequenceName = "Rh_seq", allocationSize = 1)
	@GeneratedValue(generator = "RhSeq", strategy = GenerationType.SEQUENCE)
	@JsonView(JsonViews.Common.class)
	private Long id;
	@JsonView(JsonViews.Common.class)
	private String nom;
	@JsonView(JsonViews.Common.class)
	private String prenom;

	@JsonView(JsonViews.Common.class)
	@Embedded
	private Adresse adresse;
	@JsonView(JsonViews.Common.class)
	private String telephone;
	@JsonView(JsonViews.Common.class)
	private String email;
	@JsonView(JsonViews.Common.class)
	private String login;
	private String mdp;
	private boolean enable;

	@Version
	private int version;

	public RH() {

	}

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getNom() {
		return nom;
	}

	public void setNom(String nom) {
		this.nom = nom;
	}

	public String getPrenom() {
		return prenom;
	}

	public void setPrenom(String prenom) {
		this.prenom = prenom;
	}

	public Adresse getAdresse() {
		return adresse;
	}

	public void setAdresse(Adresse adresse) {
		this.adresse = adresse;
	}

	public String getTelephone() {
		return telephone;
	}

	public void setTelephone(String telephone) {
		this.telephone = telephone;
	}

	public String getEmail() {
		return email;
	}

	public void setEmail(String email) {
		this.email = email;
	}

	public String getLogin() {
		return login;
	}

	public void setLogin(String login) {
		this.login = login;
	}

	public String getMdp() {
		return mdp;
	}

	public void setMdp(String mdp) {
		this.mdp = mdp;
	}

	public boolean isEnable() {
		return enable;
	}

	public void setEnable(boolean enable) {
		this.enable = enable;
	}

	public int getVersion() {
		return version;
	}

	public void setVersion(int version) {
		this.version = version;
	}

	@Override
	public int hashCode() {
		final int prime = 31;
		int result = 1;
		result = prime * result + ((id == null) ? 0 : id.hashCode());
		return result;
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		RH other = (RH) obj;
		if (id == null) {
			if (other.id != null)
				return false;
		} else if (!id.equals(other.id))
			return false;
		return true;
	}

}

File: FinalAngular/src/app/model/rh.ts

import {Adresse} from './adresse';
export class RH {
  private _telephone: string;
  private _email: string;
  private _login: string;
  private _mdp: string;
  private _enable: boolean;

  constructor(private _nom: string, private _prenom: string, private _adresse: Adresse, private _id?: number) {
  }
  public get id(): number {
    return this._id;
  }

  public set id(value: number) {
    this.id = value;
  }
  public get prenom(): string {
    return this._prenom;
  }

  public set prenom(value: string) {
    this._prenom = value;
  }

  public get nom(): string {
    return this._nom;
  }

  public set nom(value: string) {
    this._nom = value;
  }
  public get adresse(): Adresse {
    return this._adresse;
  }

  public set adresse(value: Adresse) {
    this._adresse = value;
  }
    public get telephone(): string {
    return this._telephone;
  }

  public set telephone(value: string) {
    this._telephone = value;
  }
    public get email(): string {
    return this._email;
  }

  public set email(value: string) {
    this._email = value;
  }
    public get login(): string {
    return this._login;
  }

  public set login(value: string) {
    this._login = value;
  }
    public get mdp(): string {
    return this._mdp;
  }

  public set mdp(value: string) {
    this._mdp = value;
  }
      public get enable(): boolean {
    return this._enable;
  }

  public set enable(value: boolean) {
    this._enable = value;
  }
}
